core/src/main/java/ai/timefold/solver/core/preview/api/domain/variable/declarative/ShadowVariableSession.java

```diff
@@ -0,0 +1,17 @@
+package ai.timefold.solver.core.preview.api.domain.variable.declarative;
+
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+@NullMarked
+public interface ShadowVariableSession {
```

Comment:
This seems similar to ConstraintVerifier, so the naming should be similar.
Does the ConstraintVerifier API have anything that is a Session?

Reply:
As far as I understand, this does not need to be in the public API. Please remove.